backend/tests/middleware/validation.test.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * @jest-environment node
+ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { Entry, EntryAnalysis, EntryConversation, Journal, User } from '../../src/models/index.js';
+import { NextFunction, Request, Response } from 'express';
+import { validateAccount, validateEntry, validateEntryAnalysis, validateEntryConversation, validateJournal, validateLogin, validateNewPassword, validateRegistration } from '../../src/middleware/validation.js';
+import ExpressError from '../../src/utils/ExpressError.js';
+
+jest.mock('../../src/utils/ExpressError.js');
+jest.mock('../../src/models/index.js', () => {
+  return {
+    User: {
+      baseJoi: jest.fn().mockReturnValue({ error: null, value: {} }),
+      passwordJoi: jest.fn().mockReturnValue({ error: null }),
+      registrationJoi: jest.fn().mockReturnValue({ error: null, value: {} }),
+      accountJoi: jest.fn().mockReturnValue({ error: null }),
+    },
+    Journal: {
+      joi: jest.fn().mockReturnValue({ error: null, value: {} }),
+    },
+    Entry: {
+      joi: jest.fn().mockReturnValue({ error: null, value: {} }),
+    },
+    EntryAnalysis: {
+      joi: jest.fn().mockReturnValue({ error: null, value: {} }),
+    },
+    EntryConversation: {
+      joi: jest.fn().mockReturnValue({ error: null, value: {} }),
+    },
+  };
+});
+
+describe('Validation Middleware Tests', () => {
+  let req: Request;
+  let res: Response;
+  let next: NextFunction;
+
+  const mockValidation = (Model: any, method: string, error = null, value = {}) => {
+    Model[method].mockReturnValue({ error, value });
```

Comment:
There's a subtle issue here, but I don't think it's going to affect things going forward, and doesn't break any of the tests currently, so I don't think you need to address it.

Using `.mockReturnValue` will override the base implementation, and carries over between tests. So if you call `mockValidation(User, 'baseJoi', null, { email: 'test@example.com' });`, every other test will return `{ email: 'test@example.com' }` until you call `mockValidation` again with a different input. You could get around this by using `mockReturnValueOnce()` instead so that it'll go back to the original `mockReturnValue` after the mock returns a value.

Reply:
Yes, very subtle and worth addressing. Thank you for pointing it out! I resolved it in the last commit 🙂